turbine/src/cluster_nodes.rs

```diff
@@ -84,12 +84,13 @@ pub struct ClusterNodes<T> {
     _phantom: PhantomData<T>,
 }
 
-type CacheEntry<T> = Option<(/*as of:*/ Instant, Arc<ClusterNodes<T>>)>;
+// Cache entries are wrapped in Arc<OnceLock<...>>, so that, when needed, only
+// one single thread initializes the entry for the epoch without holding a lock
+// on the entire cache.
+type LruCacheOnce<K, V> = RwLock<LruCache<K, Arc<OnceLock<V>>>>;
 
 pub struct ClusterNodesCache<T> {
-    // Cache entries are wrapped in Arc<RwLock<...>>, so that, when needed, only
-    // one thread does the computations to update the entry for the epoch.
-    cache: RwLock<LruCache<Epoch, Arc<RwLock<CacheEntry<T>>>>>,
+    cache: LruCacheOnce<Epoch, (/*as of:*/ Instant, Arc<ClusterNodes<T>>)>,
```

Comment:
Why not https://crates.io/crates/lazy-lru ?

Reply:
The `LruCache` here is indeed `lazy_lru::LruCache`.